### PR TITLE
Accessibility fixes

### PR DESCRIPTION
**Pull request by iainx**
Some improvements to the already outstanding accessibility support.

### CODE REVIEW

File: Xamarin.PropertyEditing.Mac/Controls/BaseEditorControl.cs [hunk 1]
```diff
@@ -19,7 +21,25 @@ public NSButton ActionButton
 			get { return actionButton; }
 		}
 
-		public event EventHandler PropertyButtonClicked;
+		event EventHandler propertyButtonClicked;
+		public event EventHandler PropertyButtonClicked {
+			add {
+				lock (propertyButtonClicked) {
+					propertyButtonClicked += value;
+					propertyButton.Hidden = false;
```
Comment (ermau) [inline Xamarin.PropertyEditing.Mac/Controls/BaseEditorControl.cs]: This doesn't seem related to accessibility changes. We won't be handling whether property buttons are visible or not this way, currently we should assume always visible.

Comment (iainx) [reply Xamarin.PropertyEditing.Mac/Controls/BaseEditorControl.cs]: Fair enough, the reason I did it was buttons that are visible and don't do anything is fairly bad for accessibility, but I can remove it

Comment (CartBlanche) [reply Xamarin.PropertyEditing.Mac/Controls/BaseEditorControl.cs]: @iainx My CustomExpression PR refactors the PropertyButton functionality into its own class. I also agree that Buttons that do nothing, in the case of ReadOnly properties, shouldn't be visible. Showing something that isn't interactive is both bad accessibility and a bad UX.

**Comment by CartBlanche** [comment]
@iainx, @ermau, @lewing I've rebased these changes off of master, which included the PropertyButton refactor. Iain could you please take a look and confirm I've done the right thing and not broken anything??
